applications/OptimizationApplication/python_scripts/responses/linear_strain_energy_response_function.py

```diff
@@ -0,0 +1,47 @@
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.OptimizationApplication as KratosOA
+from KratosMultiphysics.OptimizationApplication.utilities.optimization_info import OptimizationInfo
+from KratosMultiphysics.OptimizationApplication.execution_policies.execution_policy_decorator import ExecutionPolicyDecorator
+from KratosMultiphysics.OptimizationApplication.responses.response_function import ResponseFunction
+from KratosMultiphysics.OptimizationApplication.responses.response_function import SupportedSensitivityFieldVariableTypes
+
+
+class LinearStrainEnergyResponseFunction(ResponseFunction):
+    @classmethod
+    def GetSensitivityFieldVariables(cls) -> 'list[SupportedSensitivityFieldVariableTypes]':
+        return [Kratos.SHAPE_SENSITIVITY, KratosOA.YOUNG_MODULUS_SENSITIVITY, KratosOA.POISSON_RATIO_SENSITIVITY]
+
+    def __init__(self, model: Kratos.Model, parameters: Kratos.Parameters, optimization_info: OptimizationInfo):
+        default_settings = Kratos.Parameters("""{
+            "evaluated_model_part_names": [
+                "PLEASE_PROVIDE_A_MODEL_PART_NAME"
+            ],
+            "primal_analysis_name"      : "",
+            "perturbation_size"         : 1e-8
+        }""")
+        parameters.ValidateAndAssignDefaults(default_settings)
+
+        self.primal_analysis_execution_policy_wrapper: ExecutionPolicyDecorator = optimization_info.GetOptimizationProcess(ExecutionPolicyDecorator, parameters["primal_analysis_name"].GetString())
```

Comment:
"The response should not own the analysis; this is exactly one of the issues or shortcomings of the ShapeOptApp. Analyses are called by the optimizer, and then responses and their sensitivities are calculated. Several responses may share the same analysis."

Reply:
In my opinion, optimization should not care about analysis. Assume the following scenario.

What if a response needs to run a single analysis several times (adjoint analysis). This is typical in fluid optimizations because we have adjoint stabilization parameters which needs to be run with different values. And all of them are simulations in Kratos. Then we cannot do that anymore.

How should we enable this if optimizer runs the analysis, and it does not know which parameter to change because, it is only known by response?

Reply:
As agreed, I will remove primal analysis calls from responses. `Analysis` list in optimizer should only have primal analyses. Adjoint analyses should be integrated in to respones. I will remove analysis call frfom response.

Reply:
done :)